anndata/_core/anndata.py

```diff
@@ -594,14 +594,20 @@ def _init_as_actual(
 
     def __sizeof__(self, show_stratified=None) -> int:
         def get_size(X):
-            if issparse(X):
-                X_csr = csr_matrix(X)
+            def csr_to_bytes(X_csr):
                 return X_csr.data.nbytes + X_csr.indptr.nbytes + X_csr.indices.nbytes
+
+            if isinstance(X, h5py._hl.dataset.Dataset):
+                return np.array(X.shape).prod() * X.dtype.itemsize
+            elif isinstance(X, sparse._csr.csr_matrix):
+                return csr_to_bytes(X)
```

Comment:
Please don’t access private modules, but only publicly documented class/function locations. Those packages might internally restructure and break this code.

```suggestion
            if isinstance(X, h5py.Dataset):
                return np.array(X.shape).prod() * X.dtype.itemsize
            elif isinstance(X, (sparse.csr_matrix, sparse.csc_matrix)):
                return csr_to_bytes(X)
```

Reply:
Changes Applied. Sorry I developed this quickly using type() results.